### PR TITLE
Remove FindSort and FindSortUnscoped, accept variatic arguments in Find, FindUnscoped and FindOne instead

### LINKED ISSUE #24: [document] soft fields management should be simpler, not having specific methods

### PR DESCRIPTION
**Pull request by Soulou**
Fixes #24

### CODE REVIEW

**Comment by Soulou** [comment]
Would you rather something like:

```
document.Find(ctx, name, id, &doc, document.SortField{"name"})
```

?

File: mongo/document/document.go [hunk 1]
```diff
@@ -142,25 +136,29 @@ func WhereSortUnscoped(ctx context.Context, collectionName string, query bson.M,
...
+	}
+	err := c.Find(query).Sort(fields...).All(data)
 	if err != nil {
 		return fmt.Errorf("fail to query mongo %v: %v", query, err)
 	}
 	return nil
 }
 
-func WhereSort(ctx context.Context, collectionName string, query bson.M, data interface{}, sortFields ...string) error {
+func WhereSort(ctx context.Context, collectionName string, query bson.M, data interface{}, sortFields ...SortField) error {
```
Comment (johnsudaar) [inline mongo/document/document.go]: This is not needed anymore, we could simply use the Where method

Comment (Soulou) [reply mongo/document/document.go]: Yes true, thanks

Comment (Soulou) [reply mongo/document/document.go]: updated

**Comment by johnsudaar** [comment]
The `Where` method is missing the sort fields.

**Comment by Soulou** [comment]
Done

**Pull request merged**